combine.lua

```diff
@@ -445,20 +444,36 @@ local function stacks_add_items(stockpile, stacks, items, container, ind)
     end
 end
 
+local function unlimited_stacks(types)
+    log(4, 'Unlimited stacks\n')
+
+    if opts.unlimited_stack then
```

Comment:
It might be cleaner/clearer if this `if` statement guarded the call to this function in populate_stacks instead of the body here